### ISSUE TITLE
Bug: Can not start maybe container on raspberrypi 4

### ISSUE THREAD

**Issue by niits** [maybe-finance/maybe#1466]
**Describe the bug**

TypeError: no implicit conversion of nil into Hash (TypeError) raised when start Maybe stacks defined on `docker-compose.example.yml`

**To Reproduce**
1. Set up Maybe with Docker by following <https://github.com/maybe-finance/maybe/blob/main/docs/hosting/docker.md>

**Expected behavior**

Maybe container show be started

**What version of Maybe are you using?**
Self-hosted, v0.1.0

**What operating system and browser are you using?**

Raspberrypi 4, Ubuntu 24.04.1 LTS

**Screenshots / Recordings**

```bash
Attaching to app-1, postgres-1
postgres-1  | 
postgres-1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres-1  | 
postgres-1  | 2024-11-16 08:52:47.416 UTC [1] LOG:  starting PostgreSQL 16.5 (Debian 16.5-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres-1  | 2024-11-16 08:52:47.418 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres-1  | 2024-11-16 08:52:47.418 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-1  | 2024-11-16 08:52:47.432 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-1  | 2024-11-16 08:52:47.498 UTC [30] LOG:  database system was interrupted; last known up at 2024-11-16 08:52:11 UTC
postgres-1  | 2024-11-16 08:52:47.724 UTC [30] LOG:  database system was not properly shut down; automatic recovery in progress
postgres-1  | 2024-11-16 08:52:47.767 UTC [30] LOG:  invalid record length at 0/19123C0: expected at least 24, got 0
postgres-1  | 2024-11-16 08:52:47.767 UTC [30] LOG:  redo is not required
postgres-1  | 2024-11-16 08:52:47.837 UTC [28] LOG:  checkpoint starting: end-of-recovery immediate wait
postgres-1  | 2024-11-16 08:52:47.883 UTC [28] LOG:  checkpoint complete: wrote 3 buffers (0.0%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.016 s, sync=0.003 s, total=0.051 s; sync files=2, longest=0.002 s, average=0.002 s; distance=0 kB, estimate=0 kB; lsn=0/19123C0, redo lsn=0/19123C0
postgres-1  | 2024-11-16 08:52:47.923 UTC [1] LOG:  database system is ready to accept connections
app-1       | bin/rails aborted!
app-1       | TypeError: no implicit conversion of nil into Hash (TypeError)
app-1       | 
app-1       | Tasks: TOP => db:prepare => db:load_config
app-1       | (See full trace by running task with --trace)
app-1 exited with code 0
```

**Additional context**

**Comment by breezinstein** [comment]
I can confirm that I have the same issue

**What version of Maybe are you using?**
Self-hosted, latest image

**What operating system and browser are you using?**

Raspberrypi 4, Debian GNU/Linux 12 (bookworm)

`Attaching to app-1, postgres-1
postgres-1  | 
postgres-1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
postgres-1  | 
postgres-1  | 2024-11-16 10:15:08.964 UTC [1] LOG:  starting PostgreSQL 16.5 (Debian 16.5-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
postgres-1  | 2024-11-16 10:15:08.964 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres-1  | 2024-11-16 10:15:08.964 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres-1  | 2024-11-16 10:15:08.971 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres-1  | 2024-11-16 10:15:08.984 UTC [30] LOG:  database system was shut down at 2024-11-16 10:13:54 UTC
postgres-1  | 2024-11-16 10:15:08.998 UTC [1] LOG:  database system is ready to accept connections
app-1       | bin/rails aborted!
app-1       | TypeError: no implicit conversion of nil into Hash (TypeError)
app-1       | 
app-1       | Tasks: TOP => db:prepare => db:load_config
app-1       | (See full trace by running task with --trace)
app-1 exited with code 0`

**Comment by Xetera** [comment]
Looks like a bug with the latest version. Installing a previous hash `9522a191de1f258d76ed996ba1c84e3a0629dd80` works fine for me

**Comment by dngmns** [comment]
Same issue.

What version of Maybe are you using?
Self-hosted, latest image

What operating system and browser are you using?
Linux

**Comment by braden-y** [comment]
Same here.

What version of Maybe are you using?
Self-hosted, latest image

What operating system and browser are you using?
Debian 12 bookworm, VM, xcp-ng 8.2.1

Fixed by using image: "ghcr.io/maybe-finance/maybe:sha-9522a191de1f258d76ed996ba1c84e3a0629dd80"

Updated app to use this image version by following instructions from docs:

cd ~/docker-apps/maybe # Navigate to whatever directory you configured the app in
docker compose pull # This pulls the "latest" published image from GHCR
docker compose build app # This rebuilds the app with updates
docker compose up --no-deps -d app # This restarts the app using the newest version

Error from docker compose logs:


app-1       | bin/rails aborted!
app-1       | TypeError: no implicit conversion of nil into Hash (TypeError)
app-1       | 
app-1       | Tasks: TOP => db:prepare => db:load_config
app-1       | (See full trace by running task with --trace)
app-1 exited with code 1


results of "docker compose run app bin/rails db:prepare --trace":

[+] Creating 1/0
 ✔ Container maybe-postgres-1  Created                                                                                                                                                                                   0.0s 
[+] Running 1/1
 ✔ Container maybe-postgres-1  Started                                                                                                                                                                                   0.5s 
** Invoke db:prepare (first_time)
** Invoke db:load_config (first_time)
** Invoke environment (first_time)
** Execute environment
** Execute db:load_config
bin/rails aborted!
TypeError: no implicit conversion of nil into Hash (TypeError)
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/railtie.rb:363:in `block (2 levels) in <class:Railtie>'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:97:in `class_eval'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:97:in `block in execute_hook'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:87:in `with_execution_control'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:92:in `execute_hook'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:78:in `block in run_load_hooks'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:77:in `each'
/usr/local/bundle/ruby/3.3.0/gems/activesupport-7.2.2/lib/active_support/lazy_load_hooks.rb:77:in `run_load_hooks'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/encryption.rb:57:in `<module:ActiveRecord>'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/encryption.rb:6:in `<main>'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `require'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `block (2 levels) in replace_require'
/usr/local/bundle/ruby/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/usr/local/bundle/ruby/3.3.0/gems/zeitwerk-2.7.1/lib/zeitwerk/core_ext/kernel.rb:34:in `require'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/base.rb:313:in `<class:Base>'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/base.rb:282:in `<module:ActiveRecord>'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/base.rb:15:in `<main>'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `require'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `block (2 levels) in replace_require'
/usr/local/bundle/ruby/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/usr/local/bundle/ruby/3.3.0/gems/zeitwerk-2.7.1/lib/zeitwerk/core_ext/kernel.rb:34:in `require'
/usr/local/bundle/ruby/3.3.0/gems/activerecord-7.2.2/lib/active_record/railties/databases.rake:23:in `block (2 levels) in <main>'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:281:in `block in execute'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:281:in `each'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:281:in `execute'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:199:in `synchronize'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:199:in `invoke_with_call_chain'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:243:in `block in invoke_prerequisites'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:241:in `each'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:241:in `invoke_prerequisites'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:218:in `block in invoke_with_call_chain'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:199:in `synchronize'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:199:in `invoke_with_call_chain'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/task.rb:188:in `invoke'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:188:in `invoke_task'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:138:in `block (2 levels) in top_level'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:138:in `each'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:138:in `block in top_level'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:147:in `run_with_threads'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:132:in `top_level'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands/rake/rake_command.rb:27:in `block (2 levels) in perform'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/application.rb:214:in `standard_exception_handling'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands/rake/rake_command.rb:27:in `block in perform'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands/rake/rake_command.rb:44:in `block in with_rake'
/usr/local/bundle/ruby/3.3.0/gems/rake-13.2.1/lib/rake/rake_module.rb:59:in `with_application'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands/rake/rake_command.rb:41:in `with_rake'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/command.rb:156:in `invoke_rake'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/command.rb:73:in `block in invoke'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/command.rb:149:in `with_argv'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/command.rb:69:in `invoke'
/usr/local/bundle/ruby/3.3.0/gems/railties-7.2.2/lib/rails/commands.rb:18:in `<main>'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `require'
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:75:in `block (2 levels) in replace_require'
/usr/local/bundle/ruby/3.3.0/gems/bootsnap-1.18.4/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
/rails/bin/rails:4:in `<main>'
Tasks: TOP => db:prepare => db:load_config

**Comment by dngmns** [comment]
Fixed by using image: "ghcr.io/maybe-finance/maybe:sha-9522a191de1f258d76ed996ba1c84e3a0629dd80"

Worked for me as well. Thanks

**Comment by iloabn** [comment]
Adding some more context to this issue:
On Discord there is [a separate thread](https://discord.com/channels/1191501936281784361/1307170873735774248) about this issue. One user there wrote that it might be the Plaid integration causing this issue for self-hosted users, they wrote:
> Seems to be a problem with the new Plaid integration because I was able to start up Maybe after rolling back to the [sha-9522a191de1f258d76ed996ba1c84e3a0629dd80 tag](https://github.com/maybe-finance/maybe/pkgs/container/maybe/303736107?tag=sha-9522a191de1f258d76ed996ba1c84e3a0629dd80), which is the [second-to-last commit](https://github.com/maybe-finance/maybe/commit/9522a191de1f258d76ed996ba1c84e3a0629dd80) before the Plaid integration was merged.

(I decided against posting their discord username here publicly but just linking to the thread instead, my intention is not to keep them from receiving credit, but rather to not share their name without their consent)

**Comment by acflint** [comment]
Same issue here on my Render environment. Going back to before the Plaid integration worked. 

This is the line that is throwing `TypeError: no implicit conversion of nil into Hash (TypeError)`:
<img width="861" alt="Screenshot 2024-11-16 at 18 52 19" src="https://github.com/user-attachments/assets/189c83c4-d1f9-422b-8e6b-d7a5f98788ea">

Which is being called in the production environment config here:
<img width="745" alt="Screenshot 2024-11-16 at 18 54 20" src="https://github.com/user-attachments/assets/ad24346c-ee9a-447e-91aa-7562037fee0d">

This line was just added to the production config in the Plaid integration yesterday. So that seems like the culprit. Seems like credentials is being used to store the activerecord encryption keys, which are used to encrypt the Plaid access_tokens. But since all of our environments are using different master keys, we can't access the credentials that were committed to the repo.

I think the best solution here is to use ENV variables instead of Rails credentials to store the activerecord encryption keys, as described here: https://edgeguides.rubyonrails.org/active_record_encryption.html#setup. I've tested that on my Render environment, and it's working great. I can work on a PR to implement this tomorrow.

**Comment by zachgoll** [comment]
@acflint I think you're spot on with this one.  Given we're supporting both hosted/self-hosted, it probably makes sense to configure this with environment variables.

That said, since the only use of the encryption credentials is for the Plaid integration, it may make sense to only conditionally require them to be configured based on whether the user is attempting to self host Plaid (this may not be feasible—we're still early on trying to test run things)

That way self-hosters continue to have an easy setup while giving them the _option_ to enable encryption if needed.

**Comment by acflint** [comment]
> That said, since the only use of the encryption credentials is for the Plaid integration, it may make sense to only conditionally require them to be configured based on whether the user is attempting to self host Plaid (this may not be feasible—we're still early on trying to test run things)

That makes sense! I started a PR, but I can give it another shot with the idea of making it conditional\optional.

**Comment by zachgoll** [comment]
This is fixed via #1476 

Self hosters should not have to make any config changes.  Just upgrade to the latest version and you should be good.

**Comment by breezinstein** [comment]
> This is fixed via #1476
> 
> Self hosters should not have to make any config changes. Just upgrade to the latest version and you should be good.

Can confirm that this fixes it
